src/script/ui/state/rdkitjs/index.js

```diff
@@ -0,0 +1,104 @@
+import molfile from '../../../chem/molfile';
+import { load } from '../shared';
+import Action from '../../../editor/shared/action';
+import { fromSgroupAddition, fromSgroupDeletion } from '../../../editor/actions/sgroup';
+import Vec2 from '../../../util/vec2';
+
+/**
+ * Performs transformation on the Molecule present in Ketcher, using utilities
+ * from the RDKit JS library.
+ *
+ * @param method: the RdkitJS transformation method to call
+ * @returns {Function}
+ */
+export function rdkitJsTransform(method) {
+	return (dispatch, getState) => {
+		const state = getState();
+		const struct = state.editor.struct().clone(null, null, false, new Map());
+		const mol = Module.get_mol(molfile.stringify(struct, // eslint-disable-line no-undef
+			{ ignoreErrors: true }));
+		if (!mol.is_valid()) {
+			alert(`The specified input molecule is invalid.`);
+			return;
+		}
+		dispatch(load(doTransformation(method, mol), {
+			rescale: method === 'layout',
+			reactionRelayout: method === 'clean'
+		}));
+	};
+}
+
+
+export function rdkitJsCIP() {
+	return (dispatch, getState) => {
+		const state = getState();
+		const struct = state.editor.struct().clone(null, null, false, new Map());
+		const mol = Module.get_mol(molfile.stringify(struct, // eslint-disable-line no-undef
+			{ ignoreErrors: true }));
+		if (!mol.is_valid()) {
+			alert(`The specified input molecule is invalid.`);
+			return;
+		}
+		const res = JSON.parse(mol.get_stereo_tags());
+		dispatch(calculateCip(res));
+	};
+}
+
+export function calculateCip(result) {
+	return (dispatch, getState) => {
+		const state = getState();
+		const editor = state.editor;
+		const restruct = editor.render.ctab;
+		const atomsIterator = restruct.molecule.atoms.keys();
+		const keys = [...atomsIterator];
+		const attributes = {
+			absolute: false,
+			attached: false,
+			context: 'Atom',
+			fieldName: 'CIP_DESC',
+			fieldValue: '(R)',
+			init: true
+		};
+		const action = new Action();
+		deleteAllSGroupsWithName(restruct, action, attributes.fieldName);
+		result.CIP_atoms.forEach((a) => {
+			const [index, fieldValue] = a;
+			const atomIdx = keys[index];
```

Comment:
Could you rename this to `atomKey`?

Is a `key` a string?

Reply:
Sure, renamed.

Yes it's a string